route/build/src/base_route.f90

```diff
@@ -0,0 +1,60 @@
+MODULE base_route
+
+  ! Abstract class for routing method
```

Comment:
I suggest a little more documentation here, to say that this is the base class for which specific routing methods will be extended from. It adds a subroutine that will need to be implemented by each specific class.

Reply:
add this comments